wled00/FX_fcn.cpp

```diff
@@ -228,6 +228,7 @@ CRGBPalette16 &Segment::loadPalette(CRGBPalette16 &targetPalette, uint8_t pal) {
     case FX_MODE_GLITTER    : pal = 11; break; // rainbow colors
     case FX_MODE_SUNRISE    : pal = 35; break; // heat palette
     case FX_MODE_RAILWAY    : pal =  3; break; // prim + sec
+    case FX_MODE_2DSOAP     : pal = 11; break; // rainbow colors
```

Comment:
It may also be good to select palette using metadata.

Reply:
TBH I prefer using coded defaults over metadata for palettes (metadata is VERY useful for speed, intensity and custom sliders as they have no concept of a "default value").
For palettes, I'd personally reserve setting a specific palette via metadata only where the effect absolutely only looks good with that one palette, and even in that case we could consider setting `pal=0` in metadata and hardcoding the default palette. The main reason for that is that the user may be confused by palettes changing "by themselves", potentionally even without them noticing if they change FX while not in PC mode (and I also like to keep fxdata as small as possible)

Removed `ledsbuff` and the second inner loop respectively, they seem like a remnant of porting the FX which no longer required in WLED :)